FreeRTOS-Plus/Demo/coreHTTP_Windows_Simulator/HTTP_S3_Download_Multithreaded/DemoTasks/S3DownloadMultithreadedHTTPExample.c

```diff
@@ -0,0 +1,1142 @@
+                xResponseItem.xResponse.pBuffer = xResponseItem.ucResponseBuffer;
+
+                /* Log contents of server response. */
+                LogInfo( ( "The response task retrieved a server response from the response queue." ) );
+                LogDebug( ( "Response Headers:\n%.*s",
+                            ( int32_t ) xResponseItem.xResponse.headersLen,
+                            xResponseItem.xResponse.pHeaders ) );
+                LogDebug( ( "Response Status:\n%u",
+                            xResponseItem.xResponse.statusCode ) );
+                LogInfo( ( "Response Body:\n%.*s\n",
+                           ( int32_t ) xResponseItem.xResponse.bodyLen,
+                           xResponseItem.xResponse.pBody ) );
+
+                /* Check for a partial content status code (206), indicating a
+                 * successful server response. */
+                if( xResponseItem.xResponse.statusCode != httpexampleHTTP_STATUS_CODE_PARTIAL_CONTENT )
+                {
+                    LogError( ( "Received response with unexpected status code: %d", xResponseItem.xResponse.statusCode ) );
+                    break;
+                }
+
+                /* Increment the number of responses found on the queue. */
+                xNumResponses += 1;
+                /* Reset the wait counter every time a response is received. */
+                ulWaitCounter = 0;
+            }
+
+            /* Break if we have received all expected responses. */
+            if( xNumResponses >= xResponseCount )
+            {
+                break;
+            }
+
+            /* Break if we have been stuck waiting for a response for too long.
+             * The total wait here will be the (notification check delay + queue
+             * check delay), multiplied by `httpexampleMAX_WAIT_ITERATIONS`. For
+             * example, with a 1000 ms delay for both checks, and a maximum
+             * iteration of 5, this function will wait 10 seconds after
+             * receiving the last response before exiting the loop. */
+            if( ++ulWaitCounter > httpexampleMAX_WAIT_ITERATIONS )
+            {
+                LogError( ( "Response receive loop exceeded maximum wait time." ) );
+                break;
+            }
+            else if( prvCheckNotification( &ulNotification, httpexampleHTTP_FAILURE, pdTRUE ) != pdFALSE )
+            {
+                LogError( ( "Response task: Received error notification from the main HTTP task. Exiting task." ) );
+                break;
+            }
+        }
+    }
+
+    /* Clear this task's notifications. */
+    xTaskNotifyStateClear( NULL );
+    ulNotification = ulTaskNotifyValueClear( NULL, ~( 0U ) );
+
+    /* Notify the other tasks of completion. */
+    xTaskNotify( xMainTask, httpexampleRESPONSE_TASK_COMPLETION, eSetBits );
+    xTaskNotify( xResponseTask, httpexampleRESPONSE_TASK_COMPLETION, eSetBits );
```

Comment:
Why is this task self notifying?

Reply:
Sorry, typo on my part -- since this isn't being used, as discussed above, I will remove that line.